packages/diffhtml-rust-parser/src/parser.rs

```diff
@@ -0,0 +1,97 @@
+//extern crate wee_alloc;
+
+use tl::{Node, Parser};
+use serde_wasm_bindgen::to_value;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::{intern};
+use js_sys::{Reflect, JsString};
+use js_sys::Array;
+use js_sys::Object;
+
+// Opt for a smaller allocator to save on resources.
+//#[global_allocator]
+//static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
+
+#[wasm_bindgen(getter_with_clone)]
+pub struct VTree {
+    pub rawNodeName: String,
+    pub nodeName: String,
+    pub nodeValue: String,
+    pub attributes: Object,
+    pub childNodes: Array,
+}
+
+// Find properties, attributes, and children and populate into a VTree struct
+// which is returned.
+pub fn crawl_node(dom_node: &Node, parser: &Parser) -> VTree {
+    let attributes = Object::new();
+    let child_nodes = Array::new();
+    let node_tag = dom_node.as_tag();
+    let mut node_name = "#text".to_string();
+
+    match node_tag {
+        Some(html_tag) => {
+            node_name = html_tag.name().as_utf8_str().to_string();
+
+            // Add the attributes
+            for attr in html_tag.attributes().iter() {
+                let key = attr.0.chars().as_str();
+                let value = match attr.1 {
+                    None => { "" }
+                    Some(ref cow_str) => {
+                        cow_str.chars().as_str()
+                    }
+                };
+
+                // FIXME Look into performance improvements this is a very expensive line.
+                Reflect::set(&attributes, &JsValue::from_str(key), &JsValue::from_str(value));
```

Comment:
Caching the value returned here has noticeable performance improvements. Once I can validate that interning is working as well, there should be a much smaller strings impact.